utils/duplicate/archiver.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from copy import deepcopy
+from typing import Tuple
+
+from requests.exceptions import HTTPError
+from assertpy import assert_warn
+from biosamples_v4.api import Client as BioSamplesClient
+from ingest.api.ingestapi import IngestApi
+from submission_broker.services.biosamples import BioSamples
+
+from converter.biosamples import BioSamplesConverter
+from .loader import DuplicateLoader
+from .submission import DuplicateSubmission, HcaSubmission
+
+
+class DuplicateArchiver:
+    def __init__(self, ingest: IngestApi, biosamples: BioSamplesClient, test_biosamples: BioSamples, converter: BioSamplesConverter):
+        self.ingest = ingest
+        self.loader = DuplicateLoader(ingest)
+        self.read_biosamples = biosamples
+        self.converter = converter
+        self.write_biosamples = test_biosamples
+        logger = logging.getLogger('biosample_comparisons')
+        logger.setLevel(logging.INFO)
+        fh = logging.FileHandler("biosample_comparisons.log")
+        fh.setLevel(logging.INFO)
+        logger.addHandler(fh)
+
+    def compare_duplicate_biosample(self, biosamples_accession: str, ignored_keys):
+        submission, biomaterial_uuid, project_uuid = self.__load_project_and_biomaterial(biosamples_accession)
+        self.__send_biosample(submission, biomaterial_uuid, project_uuid)
+        self.__log_comparison(submission, biomaterial_uuid, ignored_keys)
```

Comment:
Did you consider adding `ignored_keys` to the class' state or using a Parameter Object instead of pushing a long list of arguments in all method calls?